packages/editor-ui/src/views/helpers.ts

```diff
@@ -1,47 +1,42 @@
 import { INodeUi } from "@/Interface";
 
-export const getLeftmostTopNode = (nodes: INodeUi[]): INodeUi => {
-	let leftmostTop = nodes[0];
+interface ICorners {
+	minX: number;
+	minY: number;
+	maxX: number;
+	maxY: number;
+}
 
-	nodes.forEach(node => {
-		if (node.position[0] < leftmostTop.position[0]) {
-			leftmostTop = node;
+export const getLeftmostTopNode = (nodes: INodeUi[]): INodeUi => {
+	return nodes.reduce((leftmostTop, node) => {
+		if (node.position[0] > leftmostTop.position[0] || node.position[1] > leftmostTop.position[1]) {
+			return leftmostTop;
 		}
 
-		if (node.position[0] < leftmostTop.position[0] && node.position[1] < leftmostTop.position[1]) {
-			leftmostTop = node;
-		}
+		return node;
```

Comment:
I like the reduce approach, however I think the logic is not the same anymore. Whatever the reduce function returns becomes the new accumulator, which can be reset to any node if it is not matching the condition

I think it rather has to be this:
```Javascript
	return nodes.reduce((leftmostTop, node) => {
		if (node.position[0] > leftmostTop.position[0] || node.position[1] > leftmostTop.position[1]) {
			leftmostTop = node;
		}

		return leftmostTop;
         })
```

does this function also work for an empty array or an array with just one element?

Reply:
it works for 1 element. I decided not to worry about the empty array case because in practice it should not be possible. I can handle it if you think it's needed.
logic wise, I don't see the difference here though.

Reply:
misread the guard functionality, so alright gonna approve 😊 thx for the great work!